File: python-sdk/src/boxsdk/models/shared/postoauth2token_refreshaccesstoken.py
import dataclasses
from enum import Enum

class PostOAuth2TokenRefreshAccessTokenGrantTypeEnum(str, Enum):
    REFRESH_TOKEN = "refresh_token"


@dataclasses.dataclass
class PostOAuth2TokenRefreshAccessToken:
    r"""PostOAuth2TokenRefreshAccessToken
    A request to refresh an Access Token. Use this API to refresh an expired
    Access Token using a valid Refresh Token.
    """
    
    client_id: str = dataclasses.field(metadata={'form': { 'field_name': 'client_id' }})
    client_secret: str = dataclasses.field(metadata={'form': { 'field_name': 'client_secret' }})
    grant_type: PostOAuth2TokenRefreshAccessTokenGrantTypeEnum = dataclasses.field(metadata={'form': { 'field_name': 'grant_type' }})
    refresh_token: str = dataclasses.field(metadata={'form': { 'field_name': 'refresh_token' }})
    

File: ts-sdk/src/sdk/models/shared/postoauth2tokenrefreshaccesstoken.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";


export enum PostOAuth2TokenRefreshAccessTokenGrantTypeEnum {
    RefreshToken = "refresh_token"
}


// PostOAuth2TokenRefreshAccessToken
/** 
 * A request to refresh an Access Token. Use this API to refresh an expired
 * Access Token using a valid Refresh Token.
**/
export class PostOAuth2TokenRefreshAccessToken extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "form, name=client_id;" })
  clientId: string;

  @SpeakeasyMetadata({ data: "form, name=client_secret;" })
  clientSecret: string;

  @SpeakeasyMetadata({ data: "form, name=grant_type;" })
  grantType: PostOAuth2TokenRefreshAccessTokenGrantTypeEnum;

  @SpeakeasyMetadata({ data: "form, name=refresh_token;" })
  refreshToken: string;
}
